index.d.ts

```diff
@@ -1,66 +1,257 @@
+
+/**
+ * Interface for Turnstile rendering parameters.
+ */
+export interface RenderOptions {
+	/**
+	 * Every widget has a sitekey. This sitekey is associated with the corresponding
+	 * widget configuration and is created upon the widget creation.
+	 * - Data Attribute - `data-sitekey`
+	 */
+	sitekey: string
+
+	/**
+	 * A customer value that can be used to differentiate widgets under the same
+	 * sitekey in analytics and which is returned upon validation. This can only
+	 * contain up to 32 alphanumeric characters including `_` and `-`.
+	 * - Data Attribute - `data-action`
+	 */
+	action?: string
+
+	/**
+	 * A customer payload that can be used to attach customer data to the challenge
+	 * throughout its issuance and which is returned upon validation. This can only
+	 * contain up to 255 alphanumeric characters including `_` and `-`.
+	 * - Data Attribute - `data-cdata`
+	 */
+	cData?: string
+
+	/**
+	 * Callback function invoked upon successful challenge completion.
+	 * - Data Attribute - `data-callback`
+	 * @param token - The token passed upon successful challenge.
+	 */
+	callback: (token: string) => void
+
+	/**
+	 * Callback invoked when there is an error (e.g., network error, challenge failed).
+	 * - Data Attribute - `data-error-callback`
+	 * @see [Client-side errors](https://developers.cloudflare.com/turnstile/reference/client-side-errors)
+	 */
+	'error-callback'?: () => void
+
+	/**
+	 * Execution controls when to obtain the token of the widget and can be on `"render"` (default) or on `"execute"`.
+	 * - Data Attribute - `data-execution`
+	 * @defaultValue "render"
+	 * @see [Execution modes](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#execution-modes)
+	 */
+	execution?: 'render' | 'execute'
+
+	/**
+	 * Callback invoked when the token expires and does not reset the widget.
+	 * - Data Attribute - `data-expired-callback`
+	 */
+	'expired-callback'?: () => void
+
+	/**
+	 * Callback invoked before the challenge enters interactive mode.
+	 * - Data Attribute - `data-before-interactive-callback`
+	 */
+	'before-interactive-callback'?: () => void
+
+	/**
+	 * Callback invoked when the challenge has left interactive mode.
+	 * - Data Attribute - `data-after-interactive-callback`
+	 */
+	'after-interactive-callback'?: () => void
+
+	/**
+	 * Callback invoked when a given client/browser is not supported.
+	 * - Data Attribute - `data-unsupported-callback`
+	 */
+	'unsupported-callback'?: () => void
+
+	/**
+	 * The widget theme. Can be `"light"`, `"dark"`, or `"auto"`.
+	 * - Data Attribute - `data-theme`
+	 */
+	theme?: 'light' | 'dark' | 'auto'
+
+	/**
+	 * Language to display, either `"auto"` or an ISO 639-1 two-letter language code.
+	 * - Data Attribute - `data-language`
+	 * @see [language support FAQ](https://developers.cloudflare.com/turnstile/frequently-asked-questions/#what-languages-does-turnstile-support)
+	 */
+	language?: SupportedLanguages | 'auto' | ISO
```

Comment:
I am against a blanket `ISO`-string here because the list rarely changes and [workers-types](https://workers-types.pages.dev/#Iso3166Alpha2Code) is also hardcoding a similar list

Reply:
Oh cool!  I'll yoink that ISO type and use it instead.

Reply:
> Oh cool! I'll yoink that ISO type and use it instead.

That's a country list, not a language one :P